src/compile/data/assemble.ts

```diff
@@ -307,8 +312,8 @@ export function assembleFacetData(root: FacetNode): VgData[] {
  * @param  data array
  * @return modified data array
  */
-export function assembleData(dataCompomponent: DataComponent): VgData[] {
-  let roots: SourceNode[] = vals(dataCompomponent.sources);
+export function assembleData(model: Model, dataComponent: DataComponent): VgData[] {
```

Comment:
Er, not really... It's passed to the `addIdentifierTransforms` method called at the end of this function. The `model` is needed to check the selection component to verify whether `identifier` transforms are actually needed or not (i.e., we do not add them if all selections are projected over other fields).